source/ID/concordium-did.rst

```diff
@@ -0,0 +1,613 @@
+.. _concordium_did:
+
+==================================
+Concordium DID Method Version 1.0
+==================================
+
+About
+=====
+
+The Concordium DID method specification conforms to the requirements specified in `Decentralized Identifiers (DIDs) v1.0 <w3c-did-core-v1.0_>`_ [DID] published by the W3C Credentials Community Group.
+For more information about DIDs and DID method specifications, please see the `DID Primer`_ and `DID Spec`_.
+
+Abstract
+=========
+
+Concordium is a layer-1 blockchain with identity built into the core protocol.
+We distinguish the following types of DIDs:
+
+- **Account DID** refer to accounts on the Concordium blockchain.
```

Comment:
If the idea is that we use these uniformly to refer to credentials, then I think we need one to refer to the existing identity credentials. Referring to an account is not the right thing since it can have more than one credential.

In ID 2.0 setup the proof is associated with a credential registration ID.

We can add this part to the path and use the account DID, but we need to then define that. It seems like this solution with fragments
```
did:ccd:NET:acc:ADDR#credential-1
```
is worse than just being able to refer to a credential via its ID.

Reply:
It's an interesting point.
I don't know the exact use case for account DIDs. I imagine it's for off-chain verifiable credentials signed by the issuer (with the issuer's account keys). In this case, there might be various options, I believe.  It could be an account that sings according to the whole multi-sig structure, an individual credential, or a single key.
We could add a credential id to the fragment, instead of using indices like `credential-1`. But it might be more convenient to have a separate DID  type for that. Especially if the credential IDs are unique (are they?). I thought about credentials as part of the account structure, not as independent entities, but I could be wrong.

Maybe @tschudid can also add something to this discussion?

Reply:
I am thinking pragmatically. If the dapp asks for a proof that involves the user's ID credentials, issued by an identity provider, then we need to put some metadata into the proof as to what on-chain credential this is referring to. At the moment this is just a credential registration ID directly. If we want to have a uniform way of identifying, this should now be a DID. How should it look? An account can have multiple credentials from different identities.

Credential registration IDs are unique. That is enforced by the chain.

Reply:
Yes, I agree with that. My real question was whether a ruse credential can be fetched by it's ID directly, or one still neds to know the account address. If the account address is still needed, then fragments could be a way to go.
For example
`did:ccd:acc:4LxatD2pbmKe2zJWNF1QsSVRWPVgnfLnPPyfxVEor4pohZrKzc#8113641a210da36a9ffae6e6095b9cf486ca9b44fa059aa74577b0a1c0f7052d8e71168beccf299d767f3961b33aaae2`
In this case the fragment is the value of the `credId` field that I get through gRPC.

Then the account DID document will contain the appropriate fragment
```
{
    "@context": [
      "https://www.w3.org/ns/did/v1",
      "Concordium DID URI"
    ],
    "id": "did:ccd:testnet:acc:4LxatD2pbmKe2zJWNF1QsSVRWPVgnfLnPPyfxVEor4pohZrKzc",
    "verificationMethod": [
      {
        "id": "did:ccd:testnet:acc:4LxatD2pbmKe2zJWNF1QsSVRWPVgnfLnPPyfxVEor4pohZrKzc#acc-1",
        "controller": "did:ccd:NET:acc:ADDR",
        "type": "VerifiableCondition2021",
        "blockchainAccountId": "ADDR",
        "threshold": "T",
        "conditionThreshold": [
          {
            "verificationMethod": [
              {
                "id": "did:ccd:acc:4LxatD2pbmKe2zJWNF1QsSVRWPVgnfLnPPyfxVEor4pohZrKzc#8113641a210da36a9ffae6e6095b9cf486ca9b44fa059aa74577b0a1c0f7052d8e71168beccf299d767f3961b33aaae2",
                "controller": "did:ccd:NET:acc:ADDR",
                "type": "VerifiableCondition2021",
                "threshold": "R_1",
                "conditionThreshold": [
                  {
                    "id": "did:ccd:NET:acc:ADDR#key-1-1",
                    "type": "Ed25519VerificationKey2020",
                    "controller": "did:ccd:testnet:acc:4LxatD2pbmKe2zJWNF1QsSVRWPVgnfLnPPyfxVEor4pohZrKzc",
                    "publicKeyMultibase": "fXX"
                  },
                  "...",
                  {
                    "id": "did:ccd:NET:acc:ADDR#key-N_1-1",
                    "type": "Ed25519VerificationKey2020",
                    "controller": "did:ccd:testnet:acc:4LxatD2pbmKe2zJWNF1QsSVRWPVgnfLnPPyfxVEor4pohZrKzc",
                    "publicKeyMultibase": "fYY"
                  }
                ]
              }
            ]
          },
          "...",
            ]
          }
        ]
      }
    ],
 ...
  }
```

Otherwise, if it's possible to resolve a Concordium user credential directly, just by it's ID, we should introduce a new DID type.

Something like

```
did:ccd:NET:cred:8113641a210da36a9ffae6e6095b9cf486ca9b44fa059aa74577b0a1c0f7052d8e71168beccf299d767f3961b33aaae2
```

Reply:
You can look up via a credential ID directly. What you get is an account though, an account that has or has had the credential associated with it.